region/src/test/java/org/motechproject/nms/region/ut/CsvImportTest.java

```diff
@@ -35,90 +36,92 @@
-        csvImporter.read();
+        CsvInstanceImporter<Sample> csvInstanceImporter = createCsvInstanceImporter();
+        csvInstanceImporter.read();
     }
 
     @Test
-    public void testReadSampleWhenValid() throws Exception {
+    public void testReadSampleAsMapWhenValid() throws Exception {
+        CsvMapImporter csvMapImporter = createCsvMapImporter();
         Reader reader = new StringReader("n,s,o\n12,hello,1");
-        csvImporter.open(reader, getFieldNameMapping(), getProcessorMapping());
-        Sample sample = csvImporter.read();
-        assertEquals(12, sample.getNumber());
+        csvMapImporter.open(reader, getProcessorMapping(), getFieldNameMapping());
+        Map<String, Object> record = csvMapImporter.read();
+        assertEquals(12L, record.get("number"));
+        assertEquals("hello", record.get("string"));
+        assertEquals(sampleFromDataService, record.get("object"));
+        assertNull(csvMapImporter.read());
+    }
+
+    @Test
+    public void testReadSampleAsInstanceWhenValid() throws Exception {
+        CsvInstanceImporter<Sample> csvInstanceImporter = createCsvInstanceImporter();
+        Reader reader = new StringReader("n,s,o\n12,hello,1");
+        csvInstanceImporter.open(reader, getProcessorMapping(), getFieldNameMapping());
+        Sample sample = csvInstanceImporter.read();
+        assertEquals(12L, sample.getNumber());
         assertEquals("hello", sample.getString());
         assertEquals(sampleFromDataService, sample.getObject());
-        assertNull(csvImporter.read());
+        assertNull(csvInstanceImporter.read());
     }
 
     @Test
     public void testReadSampleWithDifferentColumnsOrderWhenValid() throws Exception {
+        CsvInstanceImporter<Sample> csvInstanceImporter = createCsvInstanceImporter();
         Reader reader = new StringReader("n,o,s\n12,1,hello");
-        csvImporter.open(reader, getFieldNameMapping(), getProcessorMapping());
-        Sample sample = csvImporter.read();
+        csvInstanceImporter.open(reader, getProcessorMapping(), getFieldNameMapping());
+        Sample sample = csvInstanceImporter.read();
         assertEquals(12, sample.getNumber());
         assertEquals("hello", sample.getString());
         assertEquals(sampleFromDataService, sample.getObject());
-        assertNull(csvImporter.read());
+        assertNull(csvInstanceImporter.read());
     }
 
     @Test(expected = CsvImportDataException.class)
     public void testReadSampleWhenProcessorConstraintIsViolated() throws Exception {
+        CsvInstanceImporter<Sample> csvInstanceImporter = createCsvInstanceImporter();
         Reader reader = new StringReader("n,s,o\n12,,1");
-        csvImporter.open(reader, getFieldNameMapping(), getProcessorMapping());
-        csvImporter.read();
+        csvInstanceImporter.open(reader, getProcessorMapping(), getFieldNameMapping());
+        csvInstanceImporter.read();
     }
 
     @Test
```

Comment:
Should a test for GetBoolean be added since we'll probably want to allow true/false and y/n as booleans.